mica/archive/aca_dark/dark_cal.py

```diff
@@ -30,21 +30,34 @@ def date_to_dark_id(date):
     """
     Convert ``date`` to the corresponding YYYYDOY format for a dark cal identifiers.
 
-    :param date: any DateTime compatible format
+    :param date: any CxoTime compatible format
     :returns: dark id (YYYYDOY)
     """
-    date = DateTime(date).date
-    return date[:4] + date[5:8]
+    t = CxoTime(date)
+    shape = np.shape(t)
+    if not shape:
+        return t.date[:4] + t.date[5:8]
+    b = t.date.view((str, 1)).reshape(-1, 21)
```

Comment:
You can use `t_date.dtype.itemsize // 4` to avoid that hardcoded 21. An alternate implementation inspired by similar code in astropy is:
```
dates = np.array(['2022:001:00:01:02.123']*6).reshape(2,3)
chars = dates.view((np.uint32, dates.dtype.itemsize // 4))
ids_char = np.concatenate([chars[..., :4], chars[..., 5:8]], axis=-1)
dark_ids = ids_char.view((str, dark_ids.shape[-1])).squeeze(-1)
```